### PR TITLE
Revert "feat(amplify-python-runtime-provider): implement python runtime provider"

### PR DESCRIPTION
**Pull request by edwardfoyle**
Reverts aws-amplify/amplify-cli#3710

### CODE REVIEW

File: packages/amplify-nodejs-runtime-provider/src/utils/legacyBuild.ts [hunk 1]
```diff
@@ -10,9 +10,7 @@ export async function buildResource(request: BuildRequest): Promise<BuildResult>
 
   if (!request.lastBuildTimestamp || isBuildStale(request.srcRoot, request.lastBuildTimestamp)) {
     installDependencies(resourceDir);
-    if (request.legacyBuildHookParams) {
-      runBuildScriptHook(request.legacyBuildHookParams.resourceName, request.legacyBuildHookParams.projectRoot);
-    }
+    runBuildScriptHook(request.legacyBuildHookParams.resourceName, request.legacyBuildHookParams.projectRoot);
```
Comment (kaustavghosh06) [inline packages/amplify-nodejs-runtime-provider/src/utils/legacyBuild.ts]: Is this tied to the python provider or the platform?

File: packages/amplify-function-plugin-interface/src/index.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export type BuildRequest = {
   env: string;
   srcRoot: string;
   runtime: string;
-  legacyBuildHookParams?: {
+  legacyBuildHookParams: {
```
Comment (kaustavghosh06) [inline packages/amplify-function-plugin-interface/src/index.ts]: Is this tied to the python provider or the platform?

File: packages/amplify-nodejs-template-provider/amplify-plugin.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "name": "functionTemplate",
+  "id": "f579f2fa-35c1-49c4-950f-b588e779b4dd",
```
Comment (kaustavghosh06) [inline packages/amplify-nodejs-template-provider/amplify-plugin.json]: this is tied to the nodejs provider right? Will this affect the existing platform?

Comment (edwardfoyle) [reply packages/amplify-nodejs-template-provider/amplify-plugin.json]: No, it's not used anymore. It won't break anything with or without it.

File: packages/amplify-function-plugin-interface/src/index.ts (hunk 2 above)
Comment (edwardfoyle) [reply packages/amplify-function-plugin-interface/src/index.ts]: Platform, but it's fine either way

File: packages/amplify-nodejs-runtime-provider/src/utils/legacyBuild.ts (hunk 1 above)
Comment (edwardfoyle) [reply packages/amplify-nodejs-runtime-provider/src/utils/legacyBuild.ts]: It's the platform but it won't break anything to revert

**Pull request merged**